kart/tabular/working_copy/base.py

```diff
@@ -719,7 +719,16 @@ def all_features_diff(self, table, schema, delta_type=Delta.insert):
         Returns a diff containing all the features in the working copy.
         """
         assert delta_type in (Delta.insert, Delta.delete)
-        pk_field = schema.pk_columns[0].name
+
+        # TODO - support altering the user-provided table to add the PK column
+        # Check that the table has a primary key:
+        try:
+            pk_field = schema.pk_columns[0].name
+        except IndexError:
+            print(
```

Comment:
Just put this message inside the NotYetImplemented exception:
- this message explains everything you need to know, so no need to have the other message that you currently have in the exception below
- putting the whole message inside the exception makes things simple and avoids worrying about things like whether to print to stdout or stderr